### PR TITLE
Modify wave size to validate samples

### PR DESCRIPTION
**Pull request by mkarunan**
Updated wave size to reflect rocWMMA library
Verification successful on all supported architectures

### CODE REVIEW

**Review by cgmillette — CHANGES_REQUESTED** [review]
rocwmma::AMDGCN_WAVE_SIZE must only be used in __global__ or __device__ code.

Specifically with the launch parameters provided by the host, these need to be queried from the device (can use code from HipDevice.hpp)

**Comment by mkarunan** [comment]
> rocwmma::AMDGCN_WAVE_SIZE must only be used in **global** or **device** code.
> 
> Specifically with the launch parameters provided by the host, these need to be queried from the device (can use code from HipDevice.hpp)

Host code has been updated with hip warp size and device/global code updated with rocwmma::AMDGCN_WAVE_SIZE.

File: samples/hip_device.hpp [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        {
+        case hipWarpSize_t::Wave32:
+        case hipWarpSize_t::Wave64:
+            mWarpSize = mProps.warpSize;
+        default:;
+        }
+
+        if( mWarpSize == hipWarpSize_t::UNSUPPORTED_WARP_SIZE)
+        {
+            std::cerr << "Cannot proceed: unsupported warp sizev detected. Exiting."
```
Comment (cgmillette) [inline samples/hip_device.hpp:58]: sizev

**Review by cgmillette — COMMENTED** [review]
Please put the device code in common.hpp to simply the sample code. Otherwise looks good!

**Review by cgmillette — APPROVED** [review]
LGTM

**Pull request merged**